template/ts-template/test/Lock.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  time,
+  loadFixture,
+} from "@nomicfoundation/hardhat-toolbox/network-helpers";
+import { expect } from "chai";
+import hre, { web3 } from "hardhat";
+import { ContractExecutionError, PayableCallOptions, ContractConstructorArgs } from "web3";
+import LockContract from "../artifacts/contracts/Lock.sol/Lock"
+import { Artifact } from "hardhat/types";
+
+describe("Lock", function () {
+  // We define a fixture to reuse the same setup in every test.
+  // We use loadFixture to run this setup once, snapshot that state,
+  // and reset Hardhat Network to that snapshot in every test.
+  const deployContract = async (
+    contractName: string,
+    constructorArgs: ContractConstructorArgs<typeof LockContract.abi>,
+    txOptions: PayableCallOptions = {}
+  ) => {
+
+    let txOptionsObj = txOptions;
+
+    // use first default account if not provided explicitly for sending transaction
+    if (web3.utils.isNullish(txOptionsObj.from))
+      txOptionsObj = {
+        from: (await web3.eth.getAccounts())[0],
+        ...txOptionsObj
+      };
+
+    const artifact: Artifact = await hre.artifacts.readArtifact(contractName);
+
+    const contract = new web3.eth.Contract(artifact.abi);  // this will not have full types support 
+
+    const deployedContract = await contract.deploy({
+      data: artifact.bytecode,
+      arguments: constructorArgs
+    }).send(txOptionsObj);
+
+    deployedContract.defaultAccount = txOptionsObj.from; // this will be used for sending future transactions to contract for all .send() function calls 
+    return deployedContract;
+  }
+
+  //TODO: add  @chainsafe/hardhat-ts-artifact-plugin output support in Artifacts object in HRE and use above function for reading abi from there with full types support
```

Comment:
Similar to comment above...
```suggestion
  // TODO: add  @chainsafe/hardhat-ts-artifact-plugin output support in
  // Artifacts object in HRE and use above function for reading abi from
  // there with full types support
```